lib/jsanalyze.js

```diff
@@ -160,6 +161,22 @@ exports.analyzeJs = function analyzeJs(contents, opts) {
 	// convert the object of symbol names to an array of symbol names
 	results.symbols = Object.keys(symbols);
 
+	// transpile
+	if (opts.transpile) {
+		var presets = [ env ];
+
+		if (opts.targets) {
+			targets = [
+				[ env, { targets: opts.targets }]
+			];
+		}
+
+		contents = results.contents = babel.transform(contents, {
```

Comment:
I know this isn't specific to your changes, but I think we may want to consider some options to retain line numbers or generate debug source mappings.

If users choose to transpile and/or minify, the Studio debugger won't know how to map the original source to the generated files for setting breakpoint lines/etc.

I think we've always hacked around this in the past by basically turning off minify when debugging, but we're going to try and always transpile now (even though it may not result in any changes), so it'll be a bigger issue.

Unfortunately, I'm not sure if Babel has fixed their source mapping, as it turned out to be broken when I did the ES6 alloy changes: https://github.com/appcelerator/alloy/pull/860

The ugly/simple fix is to set `options.retainLines = true;` which jumps through hoops to keep line numbers the same, but obviously may not really "minify" a file fully if it needs to keep a bunch of empty lines around.